llvm/lib/Target/SPIRV/SPIRVUtils.cpp

```diff
@@ -251,6 +251,24 @@ SPIRV::MemorySemantics::MemorySemantics getMemSemantics(AtomicOrdering Ord) {
   llvm_unreachable(nullptr);
 }
 
+SPIRV::Scope::Scope getMemScope(const LLVMContext &Ctx, SyncScope::ID ID) {
```

Comment:
@AlexVlx If you don't mind, let's do this function as the following:

```
SPIRV::Scope::Scope getMemScope(LLVMContext &Context, SyncScope::ID Id) {
  // Named by
  // https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_scope_id We
  // don't need aliases for Invocation and CrossDevice, as we already have them
  // covered by "singlethread" and "" strings respectively (see implementation
  // of LLVMContext::LLVMContext()).
  static llvm::SyncScope::ID SubGroupSSID =
      Context.getOrInsertSyncScopeID("subgroup");
  static llvm::SyncScope::ID WorkGroupSSID =
      Context.getOrInsertSyncScopeID("workgroup");
  static llvm::SyncScope::ID DeviceSSID =
      Context.getOrInsertSyncScopeID("device");

  if (Id == SyncScope::SingleThread)
    return SPIRV::Scope::Invocation;
  if (Id == SubGroupSSID)
    return SPIRV::Scope::Subgroup;
  if (Id == WorkGroupSSID)
    return SPIRV::Scope::Workgroup;
  if (Id == DeviceSSID)
    return SPIRV::Scope::Device;
  if (Id == SyncScope::System)
    return SPIRV::Scope::CrossDevice;

  // This function assumes that known memory sync scopes may appear to be the
  // non-exhaustive list and allows to use the most conservative/safe choice as
  // the default in case of an unknown SyncScope::ID value.
  return SPIRV::Scope::CrossDevice;
}
```

Reply:
I'd say that removal of SSIDs struct from lib/Target/SPIRV/SPIRVInstructionSelector.cpp and this implementation of `getMemScope()` may be enough. Here we may initialize and refer all scopes concisely.

Reply:
I've taken this suggestion, but typed it out slightly strangely (I can re-work it if there are objections) to keep the IDs still together / allow for easy hoisting if it turns out we want to re-use them elsewhere.

Reply:
No problems, I'm quite happy with the way the suggestion is integrated, looks as good, thank you.